lib/authz/permissions_test.go

```diff
@@ -451,3 +444,90 @@ type fakeCtxChecker struct {
 func (c *fakeCtxChecker) GetAccessState(_ types.AuthPreference) services.AccessState {
 	return c.state
 }
+
+type testClient struct {
+	services.ClusterConfiguration
+	services.Trust
+	services.Access
+	services.Identity
+	types.Events
+}
+
+func newTestResources(t *testing.T) (*testClient, *services.LockWatcher, Authorizer) {
+	backend, err := memory.New(memory.Config{})
+	require.NoError(t, err)
+
+	clusterConfig, err := local.NewClusterConfigurationService(backend)
+	require.NoError(t, err)
+	caSvc := local.NewCAService(backend)
+	accessSvc := local.NewAccessService(backend)
+	identitySvc := local.NewIdentityService(backend)
+	eventsSvc := local.NewEventsService(backend)
+
+	testClient := &testClient{
+		ClusterConfiguration: clusterConfig,
+		Trust:                caSvc,
+		Access:               accessSvc,
+		Identity:             identitySvc,
+		Events:               eventsSvc,
+	}
+
+	// Set default singletons
+	ctx := context.Background()
+	testClient.SetAuthPreference(ctx, types.DefaultAuthPreference())
+	testClient.SetClusterAuditConfig(ctx, types.DefaultClusterAuditConfig())
+	testClient.SetClusterNetworkingConfig(ctx, types.DefaultClusterNetworkingConfig())
+	testClient.SetSessionRecordingConfig(ctx, types.DefaultSessionRecordingConfig())
+
+	lockSvc := local.NewAccessService(backend)
+
+	lockWatcher, err := services.NewLockWatcher(ctx, services.LockWatcherConfig{
+		ResourceWatcherConfig: services.ResourceWatcherConfig{
+			Component: "test",
+			Client:    testClient,
+		},
+		LockGetter: lockSvc,
+	})
+	require.NoError(t, err)
+
+	authorizer, err := NewAuthorizer(AuthorizerOpts{
+		ClusterName: clusterName,
+		AccessPoint: testClient,
+		LockWatcher: lockWatcher,
+	})
+	require.NoError(t, err)
+
+	return testClient, lockWatcher, authorizer
+}
+
+func createUserAndRole(client *testClient, username string, allowedLogins []string, allowRules []types.Rule) (types.User, types.Role, error) {
+	ctx := context.TODO()
```

Comment:
```suggestion
	ctx := context.Background()
```

Don't use context.TODO() unless you plan a follow up.